Mass burg assignation

Avengium opened the pull request:
# Description

This branch has a rough sketch or idea of ​​how it would be to implement a function to move burgs en masse from one burg type to another using an interface.

This uses some of the code created by Deuzeks Machina to move burgs from one type to another plus interface and modifications made by me.


# Type of change

<!-- Please put X into brackers of required option OR delete options that are not relevant -->

- [ ] Bug fix
- [x] New feature <--------------
- [ ] Refactoring / style
- [ ] Documentation update / chore
- [ ] Other (please describe)

# Versioning

Version is not updated. We should discuss if this change is a 0.001, a 0.01 a 0.1 or different.

Versioning file needs to be changed.

Avengium committed:
burgAssign code by deuzeksmckinna

Starting point for moving burgs to a 'burg group' in bulk when burgs meet certain specific criteria.
Ideal version will have several filters as written on: Discord/fmg-suggestions/Mass Burg assignation/

Avengium committed:
fixed typo on icon-layer-group

Avengium committed:
Merge branch 'Azgaar:master' into mass-burg-assignation

Avengium committed:
added UI for burg group filter

- Code for filtering burgs.
- Added new pop up menu when clicking the button "reassign burgs".

StempunkDev commented on modules/ui/burgs-overview.js:
```diff
@@ -586,6 +600,75 @@ function overviewBurgs(settings = {stateId: null, cultureId: null}) {
+    );
+
+    // Move the filtered burgs to the 'towns' group
+    filteredBurgs.forEach(b => moveBurgToGroup(b.i, 'towns'));
```
Change this line to use the selected group instead of 'towns'

StempunkDev commented on modules/ui/burgs-overview.js:
```diff
@@ -586,6 +600,75 @@ function overviewBurgs(settings = {stateId: null, cultureId: null}) {
+    let checkPopMax = burgsAssignmentCheckPopMax.checked;
+    let selectedState = burgsAssignmentSelectedState.value;
+    let selectedCulture = burgsAssignmentSelectedCulture.value;
+
```
Get the selected group here

StempunkDev reviewed (changes requested):
It could work after these changes

Avengium committed:
EvolvedExperiment checks

Thanks to EvolvedExperiment.
- New interface
- New checks
- Include/exclude.

Avengium committed:
burg assign popup explanation

- added bmaCheckInfo. A text that says: If no filter, it will move all burgs except capitals.
- added idea "bmaMoveReview" to tell what was moved.

Azgaar commented:
This UI is ... interesting. I worked quite a lot on the tool, but have close to no idea what does it do :)
Need someone to explain and make it the way that a common user can understand what is it.

Avengium commented:
Since this feature would be better implemented in another way, I'm closing this PR to focus on other things. I'll reopen it if this project continues.
